app/controllers/admin/comments/mark_as_spam_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Admin
+  module Comments
+    class MarkAsSpamController < Admin::BaseController
```

Comment:
What do you think about having this just be in `Admin::CommentsController`? 

I guess the question is if we are doing something RESTy or not. If so, we'd want to name this `Admin::Comments::SpamController` (since we'll be doing more than marking as spam, we'll be unspamming things too). But I'm leaning towards thinking we won't have anything REST-like here, I can only think of these actions right now:

1. Listing comments
2. Marking one as spam
3. Marking one as ham
4. Marking all from an ip as spam

Reply:
Hmm, and then I guess I'm dreaming of a button that says "Delete All Spam older than 30 days"

Reply:
Yes, I've been trying to be more RESTful. I think this way will force us to keep files small and specific... but OK to go either way honestly.

Reply:
Also, @sudara ... `Marking one as ham` ... you telling me `ham` is not a typo? :)

Reply:
Thinking about the REST thing.... In order for REST to make sense, we'd need identify the entity/element we are operating on in collection/singular. `MarkAsSpam` doesn't work for me as entity we are listing/creating/destroying; it's not a noun that we are going to perform actions on.

`Comment` could be the element (`Admin::CommentsController`) and we could use member/collection routes to add actions that adjust the comment's attributes https://guides.rubyonrails.org/routing.html#adding-more-restful-actions — this is my preferred route because "spam" isn't an entity, it's an attribute that we are modifying, so it makes intuitive sense.

It could also be `Admin:Comments::Spam` if you want to view `Spam` as the noun being created/destroyed. I'd argue this would end up less clear to understand and read. What would marking something as "not spam" be? Using `DELETE` for that would be like  `destroying a comment spam` which is hard to understand I think.

But maybe there's a noun that would work? In this blog post, they talk about going more granular than the data model behind the API, one of the examples is `ChangeOfAddress`: https://www.thoughtworks.com/insights/blog/rest-api-design-resource-modeling

Hmmm, something like `MarkedSpam` might work? (a bit clunky). 

The action being performed as 1) informing an external API that they mis-categorized a comment (as "spam" or "not spam") and 2) changing the visibility of the comment on the site. I dunno, to me these keep seeming like `PUT` actions on a `Comment` resource, but maybe you can find a noun that works?

Bottom line is I'm always fine with going more granular given that we "list/create/update/destroy" a entity/resource/noun (otherwise we aren't actually doing REST)

> you telling me ham is not a typo? :)

Hehe, yeah, I actually don't like "ham" as the opposite of spam, it's confusing, but I guess it's a thing? 🐷 `mark_as_spam` and `mark_as_not_spam` sound better 🍡

Reply:
Oooh, maybe `SpamReport` or `SpamStatus`? Still might be not really REST-ful because we'd actually be listing comments, not SpamReports, and we'd never update or destroy a SpamReport.... hm...

Reply:
Sorry didn't push the code earlier :) I am already convinced on member routes and `def spam/ham`